amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/SqlSessionFactoryProvider.java

```diff
@@ -103,6 +103,7 @@ public void init(Configurations config) {
     TransactionFactory transactionFactory = new JdbcTransactionFactory();
     Environment environment = new Environment("develop", transactionFactory, dataSource);
     Configuration configuration = new Configuration(environment);
+    configuration.setDefaultExecutorType(org.apache.ibatis.session.ExecutorType.BATCH);
```

Comment:
AFAIK, this configuration only affects the statement object usage, it will not affect the transaction scope.

So I don't think we can solve the lock issue with this change.